examples/nextjs-ssr/src/app/api/auth-url/route.ts

```diff
@@ -0,0 +1,29 @@
+import { NextRequest, NextResponse } from "next/server";
+import { store } from "@/lib/store";
+import { sgidClient } from "@/lib/sgidClient";
+import { v4 as uuidv4 } from "uuid";
+
+export function GET(req: NextRequest) {
+  // Generate new session ID
+  const sessionId = uuidv4();
+
+  // Retrieve state form query params
+  const state = new URL(req.url).searchParams.get("state");
```

Comment:
What's the reason for generating the state as a param instead of in the route itself?

Also I guess if we switch to PKCE we don't need to use state anymore

Reply:
i don't really understand what you mean by 'generating the state as a param instead of in the route itself'. do you mean why is the state being extracted from the `req` the way it is?

additionally, this route has been changed to `login/page.tsx` in order to more closely mimic a traditional SSR web server.

with regards to not requiring state anymore - while that is true, I think it would still be useful to illustrate how state works as some applications may like to use it (e.g. for returning users back to where they came from)

Reply:
Right, sorry that wasn't clear. What I meant was that the value for the OAuth `state` is currently being generated by the "frontend" instead of the backend, but I see now that it's being used as a way of transmitting information (the ice cream flavor), which is fair. Let's keep it as-is then!

Reply:
@kwajiehao I think it makes more sense to be param instead of routes because it is an extra optional parameters. Beside CSRF protection, anty wanted to showcase state as "save state - before and after AS" so yeah, even though we don't need it for PKCE, still using it to demonstrate save state. In this particular example, state refers to ice cream flavour.